test/v200/utils/common/parent_command_test_utils.go

```diff
@@ -0,0 +1,205 @@
+func (testDevFile *TestDevfile) createParentExecCommand() *schema.CommandParentOverride {
+
+	LogInfoMessage("Create a parent exec command :")
+	command := schema.CommandParentOverride{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+	command.Exec = &schema.ExecCommandParentOverride{}
+	testDevFile.parentCommandAdded(command)
+	return &command
+
+}
+
+// SetParentExecCommandValues randomly sets/updates exec command attributes to random values
+func (testDevFile *TestDevfile) SetParentExecCommandValues(command *schema.CommandParentOverride) {
+
+	execCommand := command.Exec
+
+	// exec command must be mentioned by a container component
+	execCommand.Component = testDevFile.GetParentContainerName()
+
+	execCommand.CommandLine = GetRandomString(4, false) + " " + GetRandomString(4, false)
+	LogInfoMessage(fmt.Sprintf("....... commandLine: %s", execCommand.CommandLine))
+
+	// If group already leave it to make sure defaults are not deleted or added
+	if execCommand.Group == nil {
+		if GetRandomDecision(2, 1) {
+			execCommand.Group = testDevFile.addParentGroup()
+		}
+	}
+
+	if GetBinaryDecision() {
+		execCommand.Label = GetRandomString(12, false)
+		LogInfoMessage(fmt.Sprintf("....... label: %s", execCommand.Label))
+	} else {
+		execCommand.Label = ""
+	}
+
+	if GetBinaryDecision() {
+		execCommand.WorkingDir = "./tmp"
+		LogInfoMessage(fmt.Sprintf("....... WorkingDir: %s", execCommand.WorkingDir))
+	} else {
+		execCommand.WorkingDir = ""
+	}
+
+	execCommand.HotReloadCapable = GetBinaryDecision()
+	LogInfoMessage(fmt.Sprintf("....... HotReloadCapable: %t", execCommand.HotReloadCapable))
+
+	if GetBinaryDecision() {
+		execCommand.Env = addParentEnv(GetRandomNumber(1, 4))
+	} else {
+		execCommand.Env = nil
+	}
+	LogInfoMessage(fmt.Sprintf("parent command updated Id: %s", command.Id))
+
+}
+
+// createParentCompositeCommand creates an empty composite command in a schema structure
+func (testDevFile *TestDevfile) createParentCompositeCommand() *schema.CommandParentOverride {
+
+	LogInfoMessage("Create a parent composite command :")
+	command := schema.CommandParentOverride{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+	command.Composite = &schema.CompositeCommandParentOverride{}
+	testDevFile.parentCommandAdded(command)
+
+	return &command
+}
+
+// SetParentCompositeCommandValues randomly sets/updates composite command attributes to random values
+func (testDevFile *TestDevfile) SetParentCompositeCommandValues(command *schema.CommandParentOverride) {
+
+	compositeCommand := command.Composite
+	numCommands := GetRandomNumber(1, 3)
+
+	for i := 0; i < numCommands; i++ {
+		execCommand := testDevFile.AddParentCommand(schema.ExecCommandType)
+		compositeCommand.Commands = append(compositeCommand.Commands, execCommand.Id)
+		LogInfoMessage(fmt.Sprintf("....... command %d of %d : %s", i, numCommands, execCommand.Id))
+	}
```

Comment:
Yes, it is redundant.  When I created these parent* functions, I copied over the existing utility functions and changed the types to the corresponding parent overrides.  I deleted `SetParentCompositeCommandCommands` because it's not being called from anywhere